src/patch_parser_test.py

```diff
@@ -136,6 +136,43 @@ def test_one_deleted_line(self):
         self.assertEqual(line_map.map(11), ('file', 2), msg=repr(line_map))  # line 2, control assert
         self.assertEqual(line_map.map(13), ('fileb', 4), msg=repr(line_map))  # line 4 (deleted but shows in b file)
 
+    def test_many_deleted_lines(self):
+        raw_patch = '''
+                    Commit message goes here.
+
+                    ---
+                     file | 3 ++-
+                     1 file changed, 2 insertions(+), 1 deletion(-)
+
+                    diff --git a/file b/file
+                    index fa2da6e55caa..1fc93eb38351 100644
+                    --- a/file
+                    +++ b/file
+                    @@ -2,7 +2,8 @@ line 1
+                     line 2  # this is line 11 in raw_patch
+                     line 3 - 12
+                     - line 4 13
+                     - line 5 14
+                     - line 6 15
+                     - line 7 16
+                     line 8 (then 4) -17
+                     line 9 (then 5) -18
+                     line 10  (then 6) # this is line 19 in raw_patch
+
+                    base-commit: 235360eb7cd778d7264c5e57358a3d144936b862
+                    --
+                            '''.strip()
+        line_map = patch_parser._parse_git_patch(raw_patch)
+        self.assertEqual(line_map.map(11), ('file', 2), msg=repr(line_map))  # line 2 (11), control assert
+        self.assertEqual(line_map.map(12), ('file', 3), msg=repr(line_map))  # line 3 (12), control assert
+        self.assertEqual(line_map.map(13), ('fileb', 4), msg=repr(line_map))  # Lines 13 - 16 are deleted and should
+        self.assertEqual(line_map.map(14), ('fileb', 5), msg=repr(line_map))  # show on 'fileb' with their corresponding
```

Comment:
optional: you seem to have line-wrapped the comments, but this one word sticks out.
I think we can just move it into the comment below and everything should fit.